ci.sh

```diff
@@ -195,6 +214,38 @@ if [ "$PROVIDER_NAME" = "coverage" ]; then
 fi
 
 echo "Build test"
+
+if [ "$PROVIDER_NAME" = "all" ]; then
+    # We test that everything in the service still builds with the Rust compiler
+    # versions that we support.
+    rustup update
+
+    # Latest version in Fedora 33
+    rustup toolchain install 1.46.0
+    RUST_BACKTRACE=1 cargo +1.46.0 check --release $FEATURES
+
+    # Latest version in openSUSE Leap 15.3
+    rustup toolchain install 1.43.1
+    RUST_BACKTRACE=1 cargo +1.43.1 check --release $FEATURES
+
+    # Latest stable
+    rustup toolchain install stable
+    RUST_BACKTRACE=1 cargo +stable check --release $FEATURES
```

Comment:
Are we sure we want to do this? We could run into problems like the one with `picky-asn1`. It's not like someone using Fedora or openSUSE couldn't get the latest Rust compiler with `rustup`...

Reply:
This is basically to avoid things like https://github.com/parallaxsecond/parsec/issues/409 happening again 😕 I think people there might just want to use the version available without internet access.

Reply:
@ggardet @nullr0ute

In this PR we add build tests on CI with specific versions of the Rust compiler we need to track to make sure that Parsec compiles on Fedora and openSUSE.

I wanted to ask you where can we find the latest version of Rust that we should test?

For openSUSE, I used the comment on the issue linked above, is there a place where we can check where the Rust version has been updated?
For Fedora, I used the "base" version of Fedora 33 [here](https://rpms.remirepo.net/rpmphp/zoom.php?rpm=rust), not sure if it's a good ressource. Would the "updates" one be fine?

In other PRs we might introduce changes that would break on old versions of Rust so we would like to make sure that we track the *latest* version of Rust that you need to package Parsec in the distributions!

Reply:
So Fedora pretty actively updates the rust toolchain in all actively maintained releases, for example we have 1.52.1 across all releases currently. RHEL-8 currently has 1.49.0.

To test this in a container with Fedora you can do:
FROM fedora:latest
RUN dnf install -y cargo

For RHEL you can do:
FROM redhat/ubi8
RUN yum install -y cargo

And that will give you the latest stable containers with rust/cargo installed

Reply:
For openSUSE, you can track:
* Tumbleweed (the rolling release): https://build.opensuse.org/package/show/openSUSE:Factory/rust which is at `1.51.0` atm
* Leap 15.3 (the current stable release): https://build.opensuse.org/package/show/openSUSE:Leap:15.3/rust which is at `1.43.0`

You can of course use containers for openSUSE Tumbleweed or openSUSE Leap 15.3 for x86_64 and/or aarch64.

Reply:
I think the question isn't "what release of the compiler is available, packaged, for the distributions?", but "what release of the compiler is used for generating Rust-based packages?" The numbers could very well be the same, but I'm not sure the initial phrasing of "we need to track to make sure that Parsec compiles on Fedora and openSUSE" was accurate.

Reply:
As the Rust compiler is meant to be stable, I think it's enough to try the compilation with the oldest Rust version used to compile Parsec in the distributions we support. In that case I think it would be `1.43.0`.

We can manually update that from time to time.